File: store/src/main/java/com/example/store/Service/supplierManagement/SupplierService.java
package com.example.store.Service.supplierManagement;

import com.example.store.DTO.supplierManagement.SupplierDTO;
import com.example.store.Model.supplierManagement.Supplier;

import java.util.List;

public interface SupplierService {

    Supplier saveSupplier(SupplierDTO supplier);

    Supplier findSupplierById(Long supplierId);

    List<Supplier> fetchSupplierList();

    void deleteSupplierByID(Long supplierId);
}

File: Frontend/src/app/core/services/supplierManagment/supplier.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { Supplier } from '../../../shared/models/SupplierManagement/Suplier.model';
import { SupplierDTO } from '../../../shared/models/dto/supplierManagementDTO/supplier.dto';

@Injectable({
  providedIn: 'root'
})
export class SupplierService {

  private apiUrl = 'http://localhost:8080/api/supplier';
  private http = inject(HttpClient);

  constructor() { }

  getSuppliers():Observable<Supplier[]>{
    return this.http.get<Supplier[]>(`${this.apiUrl}/all`);
  }

  addSupplier(supplier: SupplierDTO):Observable<Supplier>{
    console.log("Supplier service is called to add a new supplier");
    return this.http.post<Supplier>(`${this.apiUrl}/suppliers`,supplier);
  }

}
